### PR TITLE
Moved certain classes to AdminBundle

### LINKED ISSUE #14605: Move certain classes to `AdminBundle`
```php
Pimcore\Event\Admin\*
Pimcore\Event\AdminEvents
AssetDeleteInfoEvent
DataObjectDeleteInfoEvent
DocumentDeleteInfoEvent
ElementDeleteInfoEventInterface
ElementDeleteInfoEventTrait
Pimcore\Helper\Dashboard
Pimcore\Helper\ImageChart
Pimcore\Perspective\Config
Pimcore\CustomView\Config
\Pimcore\Twig\Extension\AdminExtension
\Pimcore\Tool::getLanguageFlagFile()
\Pimcore\Model\GridConfig*
\Pimcore\Model\GridConfigFavourite*
\Pimcore\Model\GridConfigShare*
Pimcore\DataObject\GridColumnConfig\*
```

`\Pimcore\Workflow\ActionsButtonService` is kind of special, there we also need to split `\Pimcore\Bundle\CoreBundle\EventListener\WorkflowManagementListener` and move the admin related stuff into the admin bundle. 

`\Pimcore\Event\BundleManagerEvents` is related to #14540

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #14605

### CODE REVIEW

File: bundles/AdminBundle/src/Controller/Admin/MiscController.php [hunk 1]
```diff
@@ -374,7 +375,7 @@ public function iconListAction(Request $request, ?Profiler $profiler): Response
                 $languageOptions[] = [
                     'language' => $short,
                     'display' => $translation . " ($short)",
-                    'flag' => \Pimcore\Tool::getLanguageFlagFile($short, true),
+                    'flag' => \Pimcore\Bundle\AdminBundle\Tool::getLanguageFlagFile($short, true),
```
Comment (mattamon) [inline bundles/AdminBundle/src/Controller/Admin/MiscController.php]:
```suggestion
                    'flag' => AdminTool::getLanguageFlagFile($short, true),
```

File: bundles/AdminBundle/src/DataObject/GridColumnConfig/Operator/Factory/DefaultOperatorFactory.php [hunk 2]
```diff
@@ -15,9 +15,9 @@
  *  @license    http://www.pimcore.org/license     GPLv3 and PCL
  */
 
-namespace Pimcore\DataObject\GridColumnConfig\Operator\Factory;
+namespace  Pimcore\Bundle\AdminBundle\DataObject\GridColumnConfig\Operator\Factory;
```
Comment (mattamon) [inline bundles/AdminBundle/src/DataObject/GridColumnConfig/Operator/Factory/DefaultOperatorFactory.php]:
```suggestion
namespace Pimcore\Bundle\AdminBundle\DataObject\GridColumnConfig\Operator\Factory;
```

File: lib/Event/ReportEvents.php [hunk 3]
```diff
@@ -14,16 +14,16 @@
...
     /**
      * The SAVE_SETTINGS event is triggered when reports settings are saved
      *
-     * @Event("Pimcore\Event\Admin\Report\SettingsEvent")
+     * @Event("Pimcore\Event\Report\SettingsEvent")
      *
      * @var string
      */
-    const SAVE_SETTINGS = 'pimcore.admin.reports.save_settings';
+    const SAVE_SETTINGS = 'pimcore.reports.save_settings';
```
Comment (mattamon) [inline lib/Event/ReportEvents.php:28]: We should mention this somewhere, that the name changed. You have to change your Listener accordingly

**Review by mattamon — CHANGES_REQUESTED** [review]
I checked for an easy way to deprecate all the class aliases for the future but did not find any. 
Would be nice though if e.g. the class AdminEvents in the old namespace would be shown as deprecated for code completion.

File: lib/Event/ReportEvents.php (hunk 3 above)
Comment (dvesh3) [reply lib/Event/ReportEvents.php:28]: Added upgrade notes.

**Comment by dvesh3** [comment]
> I checked for an easy way to deprecate all the class aliases for the future but did not find any. Would be nice though if e.g. the class AdminEvents in the old namespace would be shown as deprecated for code completion.

@mattamon we could keep the old classes in the core (extending from new classes) with deprecation tag and remove the bc layer autoloading with `class_alias()`. wdyt?

**Comment by mattamon** [comment]
> > I checked for an easy way to deprecate all the class aliases for the future but did not find any. Would be nice though if e.g. the class AdminEvents in the old namespace would be shown as deprecated for code completion.
> 
> @mattamon we could keep the old classes in the core (extending from new classes) with deprecation tag and remove the bc layer autoloading with `class_alias()`. wdyt?

Hmm, that would be possible but I don't find it very practical tbh to keep unused code just for the sake of depreciation. 
I even found a jetbrains issue for this. 

https://youtrack.jetbrains.com/issue/WI-34715/Option-to-mark-a-class-alias-as-deprecated but it's from 6 years ago. 

Maybe we should just add a deprecation php doc above the aliases and be done with it. At least you could search for it right?

**Comment by dvesh3** [comment]
> Hmm, that would be possible but I don't find it very practical tbh to keep unused code just for the sake of depreciation. I even found a jetbrains issue for this.
> 
I didn't mean to duplicate classes. I mean to keep the old class extending from the new ones with deprecation like this:

```php
namespace Pimcore\Event;

/**
 * @deprecated use \Pimcore\Bundle\AdminBundle\Event\AdminEvents instead.
 */
final class AdminEvents extends \Pimcore\Bundle\AdminBundle\Event\AdminEvents
{
}
```

> https://youtrack.jetbrains.com/issue/WI-34715/Option-to-mark-a-class-alias-as-deprecated but it's from 6 years ago.
> 
> Maybe we should just add a deprecation php doc above the aliases and be done with it. At least you could search for it right?

I tried already, it doesn't work.

**Comment by mattamon** [comment]
> > Hmm, that would be possible but I don't find it very practical tbh to keep unused code just for the sake of depreciation. I even found a jetbrains issue for this.
> 
> I didn't mean to duplicate classes. I mean to keep the old class extending from the new ones with deprecation like this:
> 
> ```
> namespace Pimcore\Event;
> 
> /**
>  * @deprecated use \Pimcore\Bundle\AdminBundle\Event\AdminEvents instead.
>  */
> final class AdminEvents extends \Pimcore\Bundle\AdminBundle\Event\AdminEvents
> {
> }
> ```
> 
> > https://youtrack.jetbrains.com/issue/WI-34715/Option-to-mark-a-class-alias-as-deprecated but it's from 6 years ago.
> > Maybe we should just add a deprecation php doc above the aliases and be done with it. At least you could search for it right?
> 
> I tried already, it doesn't work.

But we can't do this in one file right? (PSR conflict) This was my concern like we now have to keep all the old classes, but they are actually empty just so we can deprecate them.  I don't like that tbh. 

What I mean with the class_aliases deprecation is, just to put the docs there, if it does not work, but we could at least search for deprecated stuff that way in the IDE

**Comment by brusch** [comment]
I'd just move the classes and do not provide a BC layer. It's a major release anyway ...

**Comment by dvesh3** [comment]
@brusch we should either provider a BC layer here or FC layer on 10.6 to keep supporting Pimcore bundles on version 10 & 11. Otherwise it will be an overhead to check version specific classes(events specially) in bundles.

**Comment by brusch** [comment]
@dvesh3 that's true 👍

**Review by mattamon — APPROVED** [review]
LGTM

**Review by mattamon — APPROVED** [review]

**Pull request merged**